src/components/FeedList/FeedList.tsx

```diff
@@ -0,0 +1,79 @@
+import React, { useState, useEffect } from 'react';
+import InfiniteScroll from 'react-infinite-scroll-component';
+import { CircularProgress } from '@material-ui/core';
+import { useGetPosts } from '../../actions/useGetPosts';
+import { useGetUserFollows } from '../../actions/useGetUserFollows';
+import { postsFilter } from '../../actions/PostsFilter';
+import { FeedPost, FeedPostProps } from '../FeedPost/FeedPost';
+import { Loader } from '../Loader/Loader';
+import { Error } from '../Error/Error';
+import './FeedList.scss';
+
+export const FeedList: React.FC = () => {
+  const [posts, setPosts] = useState<FeedPostProps[]>([]);
+  const [index, setIndex] = useState(5);
+  const [hasMore, setHasMore] = useState(true);
+
+  const { data: userFollows, isLoading: isFollowsLoading, isError: isFollowsError } = useGetUserFollows();
+  const { data: allPosts, isLoading: isPostsLoading, isError: isPostsError } = useGetPosts();
+  const userPosts = postsFilter(userFollows, allPosts);
+
+  useEffect(() => {
+    if (userFollows.length && allPosts.length) {
```

Comment:
you could split it to 2 functions

 if (userFollows.length && allPosts.length) {
      setPosts(userPosts.slice(0, 5));
    }

    if (userPosts.length <= 5 && allPosts.length) {
      setHasMore(false);
    }